src/app/clusters/basic-information/BasicInformationCluster.cpp

```diff
@@ -247,175 +254,194 @@ inline CHIP_ERROR ReadProductAppearance(DeviceInstanceInfoProvider * deviceInfoP
 
-    ReturnErrorOnFailure(aDecoder.Decode(location));
+    AttributePersistence persistence(*mContext->attributeStorage);
 
-    bool isValidLength = location.size() == kExpectedFixedLocationLength;
-    if (!isValidLength)
+    switch (request.path.mAttributeId)
     {
-        ChipLogError(Zcl, "Invalid country code: '%s'", NullTerminated(location).c_str());
-        return CHIP_IM_GLOBAL_STATUS(ConstraintError);
+    case Location::Id: {
+        CharSpan location;
+        ReturnErrorOnFailure(decoder.Decode(location));
+        VerifyOrReturnError(location.size() == kFixedLocationLength, Protocols::InteractionModel::Status::ConstraintError);
+        return DeviceLayer::ConfigurationMgr().StoreCountryCode(location.data(), location.size());
     }
+    case NodeLabel::Id: {
+        CharSpan label;
+        ReturnErrorOnFailure(decoder.Decode(label));
+        VerifyOrReturnError(mNodeLabel.SetContent(label), Protocols::InteractionModel::Status::ConstraintError);
+        return persistence.StoreString(request.path, mNodeLabel);
+    }
+    case LocalConfigDisabled::Id:
+        return persistence.DecodeAndStoreNativeEndianValue(request.path, decoder, mLocalConfigDisabled);
+    default:
+        return Protocols::InteractionModel::Status::UnsupportedWrite;
+    }
+}
 
-    return DeviceLayer::ConfigurationMgr().StoreCountryCode(location.data(), location.size());
+CHIP_ERROR BasicInformationCluster::Attributes(const ConcreteClusterPath & path,
+                                               ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder)
+{
+#define OPTIONAL_ATTR(name) { mEnabledOptionalAttributes.Has(OptionalBasicInformationAttributes::k##name), name::kMetadataEntry }
+
+    AttributeListBuilder::OptionalAttributeEntry optionalAttributes[] = {
+        { !mEnabledOptionalAttributes.Has(OptionalBasicInformationAttributes::kDisableMandatoryUniqueIDOnPurpose),
+          UniqueID::kMetadataEntry },
+        OPTIONAL_ATTR(ManufacturingDate),
+        OPTIONAL_ATTR(PartNumber),
+        OPTIONAL_ATTR(ProductURL),
+        OPTIONAL_ATTR(ProductLabel),
+        OPTIONAL_ATTR(SerialNumber),
+        OPTIONAL_ATTR(LocalConfigDisabled),
+        OPTIONAL_ATTR(Reachable),
+        OPTIONAL_ATTR(ProductAppearance),
+    };
+
+    AttributeListBuilder listBuilder(builder);
+
+    return listBuilder.Append(Span(kMandatoryAttributes), Span(optionalAttributes));
 }
 
-class PlatformMgrDelegate : public DeviceLayer::PlatformManagerDelegate
+CHIP_ERROR BasicInformationCluster::Startup(ServerClusterContext & context)
 {
-    void OnStartUp(uint32_t softwareVersion) override
+    ReturnErrorOnFailure(DefaultServerCluster::Startup(context));
+    if (PlatformMgr().GetDelegate() == nullptr)
     {
-        MATTER_TRACE_INSTANT("OnStartUp", "BasicInfo");
-        // The StartUp event SHALL be emitted by a Node after completing a boot or reboot process
-        ChipLogDetail(Zcl, "Emitting StartUp event");
```

Comment:
The detail logging here and in OnShutDown seems to have gotten lost.  Please put it back. 

The error logging on failure to generate the event also got lost.... I am not sure how useful that is in practice, but it's odd that it was just summarily removed.

Reply:
I can add it back, however realistically I believe this is noise: we already log a lot of things and we have the trace instant as well as tons of logging on every action.

For the logging of events, the idea was that we essentially never can recover in any meaningful way from event generation failures, so logging is now done by default in https://github.com/project-chip/connectedhomeip/blob/master/src/app/data-model-provider/EventsGenerator.cpp#L42

The only choice this code would have is to check if the optional event number is missing (meaning error) and handle that ... but the handling would be log only and that is already done.